src/libs/actions/BankAccounts.js

```diff
@@ -652,18 +661,8 @@ function setupWithdrawalAccount(data) {
     }
 
     API.BankAccount_SetupWithdrawal(newACHData)
-        /* eslint-disable arrow-body-style */
-        .then((response) => {
-            // Without this block, we can call merge again with the achData before this merge finishes, resulting in
-            // the original achData overwriting the data we're trying to set here. With this block, we ensure that the
-            // newACHData is set in Onyx before we call merge on the reimbursementAccount key again.
-            return Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {
-                loading: false,
-                achData: {...newACHData},
-            })
-                .then(() => Promise.resolve(response));
-        })
         .then((response) => {
+            Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {loading: false, achData: {...newACHData}});
```

Comment:
I'm admittedly a little nervous about this change. But only because I don't fully understand why the code we are removing was necessary in the first place.

The original comment says:

>Without this block, we can call merge again with the achData before this merge finishes, resulting in the original achData overwriting the data we're trying to set here.

But I'm not sure which additional call to `merge()` it's referring to. In addition, any merges made in this execution context should get batched together and resolve as one single "merge" on the Onyx side.

Putting that aside for now... is this change necessary in order to get the new `subStep` logic to work properly?

Reply:
Hmm did we talk about this already maybe? Sorry I don't think I really got it the first time 😄

Reply:
I think this is the change that fixes the whole problem.

Initially my solution was different:
- Avoid double fetch of withdrawal account information (removed from PR)
- Read `subStep` directly from Onyx instead of copying it to local state.

After more debugging, I saw that the commented code here was causing the `BankAccountStep` component to be rendered before the whole information was available. Rendering the `BankAccountStep` before the right time, caused it to get an incorrect subStep copied in its local state.

I think that If you remove this piece of code that the PR is intending to remove in `BankAccounts.js`, it is not necessary any more to read the `subStep` directly from Onyx, but I left this change there anyway because I can help preventing future bugs.

I can do more testing around this if you think it would be good.

Reply:
I tried to replicate the problem @Jag96 's code was fixing and it doesn't seem to be happening anymore. I'm thinking that it probably got fixed by some fix in Onyx. Another possibility is that it is not happening anymore because we introduced this "draft save" mechanism and I think we give priority to reading from there.

Reply:
> But I'm not sure which additional call to merge() it's referring to.

I should've specified, I don't see it now and I can't reproduce so not sure which other merge was getting batched. Maybe it was [this one](https://github.com/Expensify/App/pull/4477/files#diff-448c3537222b5e687caa8f06585ace6feae03d751f243e1590383fb6a33693e8L475).

> In addition, any merges made in this execution context should get batched together and resolve as one single "merge" on the Onyx side.

Merge 1 set the new bankAccountNumber, Merge 2 didn't set any bankAccountNumber, and when it resolved into one merge it was using the bankAccountNumber from Merge 2 which was the original one.

> I'm thinking that it probably got fixed by some fix in Onyx.

We should be able to test this pretty easily by calling Onyx.merge twice and seeing if the value set in the first merge is overwritten by the original value if the 2nd merge doesn't set that value.

Reply:
Ok sounds like this change should be fine. I tested a bunch and don't really see anything not working differently from `main`.